### PR TITLE
Add support for plugin inline feature

### LINKED ISSUE #37: Add support for plugin inline feature
Based yesterday's WG call discussion please the support for plugin inline.

In a nutshell it enables plugin to be included in the `kn` binary during build time therefore allowing custom distributions of kn and removing the complexity of using kn and plugins together in CIs, dev env etc.

More details can be found in kn's [docs](https://github.com/knative/client/tree/master/docs/plugins) and an example of such plugin is [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka/blob/master/plugin/plugin.go).

/assign @dsimansk

### PR DESCRIPTION
**Pull request by dsimansk**
# Changes
- :gift: Add support for plugin inline feature
- ~~[Fix](https://github.com/knative-sandbox/kn-plugin-admin/commit/7da6ae328f18f74c1d2a6b6f5d93133c269b0bda) unit test - this might be not necessary, but it was failing for me locally~~

Fixes #37

### CODE REVIEW

**Comment by navidshaikh** [comment]
> Fix unit test - this might be not necessary, but it was failing for me locally

are you running golang 1.15 locally?

**Comment by dsimansk** [comment]
> > Fix unit test - this might be not necessary, but it was failing for me locally
> 
> are you running golang 1.15 locally?

Yep, that's an issue apparently. I've reverted the "fix" commit.

**Review by navidshaikh — APPROVED** [review]
/lgtm
/approve
/hold for others to take a look

File: vendor/knative.dev/client/pkg/kn/plugin/stat_windows.go
Comment (maximilien) [inline vendor/knative.dev/client/pkg/kn/plugin/stat_windows.go:21]: Don’t you have to put `//+build windows` or similar to this file?

Comment (maximilien) [reply vendor/knative.dev/client/pkg/kn/plugin/stat_windows.go:21]: I know this is vendored... but perhaps a question for @rhuss ?

**Review by maximilien — APPROVED** [review]
Thanks for contribution @dsimansk

Couple minor comments. Wondering also if there is a way to test this? Maybe a common e2e tests for plugins. If we got hat route let’s make sure to link in an issue here.

Cheers :)

File: vendor/knative.dev/client/pkg/kn/plugin/stat_windows.go
Comment (navidshaikh) [reply vendor/knative.dev/client/pkg/kn/plugin/stat_windows.go:21]: @maximilien : Nope. The build constraint can also be achieved by file suffixes as you can see `_windows.go`.

Comment (maximilien) [reply vendor/knative.dev/client/pkg/kn/plugin/stat_windows.go:21]: OK @navidshaikh cool cool. I thought you needed both. Thx.

File: plugin/plugin.go
Comment (zhanggbj) [inline plugin/plugin.go:1]: Referred to latest [Knative/Serving repo](https://github.com/knative/serving/blob/3b6b40aec07654c1d2d411397122ae279a4ccfe4/pkg/reconciler/gc/reconciler.go#L2), the Copyright is 
```
Copyright 2020 The Knative Authors
```
Please help to fix it, and I've updated others in https://github.com/knative-sandbox/kn-plugin-admin/pull/41

Comment (dsimansk) [reply plugin/plugin.go]: Done in https://github.com/knative-sandbox/kn-plugin-admin/pull/38/commits/4b20ec46ae2bd73190a611948f0a358f087941ea. Sorry that it took me so long to fix it. :)

**Comment by dsimansk** [comment]
@maximilien @rhuss per WG call yesterday the PR is ready. I'll remove hold because it requires fresh `/lgtm` anyway.

/unhold

**Pull request merged**
